### DISCUSSION TITLE
Would there be a built-in reader?

### DISCUSSION THREAD

**Discussion by adriangalilea** [ymgyt/syndicationd#117] [Q&A]
I do not like switching, I specifically choose the cli to avoid the browser.

So, will there be a built-in reader?

Alternatively, is there any browser you recommend for in-terminal reading?

**Comment by ymgyt** [comment]
I also prefer to do as many things as possible using the terminal/CLI. 
Unfortunately, at the moment, the built-in reader is not implemented in syndicationd, and reading RSS feeds assumes using a browser. If there is a good way to render HTML in the terminal, I would like to incorporate it. 
Additionally, I am not aware of any browser suitable for in-terminal reading. If you find a good one, please share it with me!

**Reply by ymgyt** [reply]
I was informed about a text browser called w3m that operates in the terminal.
I am considering trying to integrate it to view articles within the terminal. Thank you for starting the discussion!

https://github.com/ymgyt/syndicationd/issues/118

**Reply by adriangalilea** [reply]
> I was informed about a text browser called w3m that operates in the terminal.

I'm also aware of browsh, not sure which is better, but I'd say the way to do it would be to read the `$BROWSER` env variable, which is a widely adopted method, and would respect the user's preferences for the desired cli browser.

Also consider having a config file, also respecting `$XDG_CONFIG_.../syndicationd/` as a default for the config path. In Go there's [this library](https://github.com/adrg/xdg) that [I use](https://github.com/adriangalilea/xdg-dirs) which you may be able to leverage or find a similar one for rust. Will be helpful for consolidating some settings.

> I am considering trying to integrate it to view articles within the terminal. Thank you for starting the discussion!

I believe the browser approach may be trickier than initially thought, unless you can embed a subshell inside your program for the reading the `$BROWSER` program (if set).

As an alternative, simply rendering the text may suffice, if the text should be scrapped, something like [newspaper4k](https://github.com/AndyTheFactory/newspaper4k) can be used.

Glad you think likewise, loving the program so far, gotta be one of the best designed TUI's I came across.

**Reply by ymgyt** [reply]
> I'm also aware of browsh, not sure which is better, but I'd say the way to do it would be to read the $BROWSER env variable, which is a widely adopted method, and would respect the user's preferences for the desired cli browser.

Thank you for the suggestion. I will try to configure it to read the `$BROWSER` environment variable.

> Also consider having a config file, also respecting $XDG_CONFIG_.../syndicationd/ as a default for the config path. In Go there's [this library](https://github.com/adrg/xdg) that [I use](https://github.com/adriangalilea/xdg-dirs) which you may be able to leverage or find a similar one for rust. Will be helpful for consolidating some settings.

In the current implementation, we are using [directories-rs](https://github.com/dirs-dev/directories-rs), which is expected to respect the XDG-related environment variables.

> I believe the browser approach may be trickier than initially thought, unless you can embed a subshell inside your program for the reading the $BROWSER program (if set).

> As an alternative, simply rendering the text may suffice, if the text should be scrapped, something like [newspaper4k](https://github.com/AndyTheFactory/newspaper4k) can be used.

Thank you for the suggestion. I will try various methods.

> Glad you think likewise, loving the program so far, gotta be one of the best designed TUI's I came across.

Thank you for trying out the program. I'm glad to receive your feedback.

**Reply by adriangalilea** [reply]
> In the current implementation, we are using [directories-rs](https://github.com/dirs-dev/directories-rs), which is expected to respect the XDG-related environment variables.

I see, sadly that won't work on my setup, since they do not pick up on XDG paths on Mac, which is what I'm using as I prefer them over Mac default ones, but no worries it's fine.

**Reply by ymgyt** [reply]
Instead of using a web browser, I tried calling w3m to view the feed on the terminal
The transition between synd and w3m seemed to work well.

![demo](https://github.com/user-attachments/assets/4873cb3f-13b2-46ee-a63b-b34bc44071a3)

**Reply by adriangalilea** [reply]
> Instead of using a web browser, I tried calling w3m to view the feed on the terminal
> 
> The transition between synd and w3m seemed to work well.
> 
> 
> 
> ![demo](https://github.com/user-attachments/assets/4873cb3f-13b2-46ee-a63b-b34bc44071a3)
> 
> 

That looks lovely.

W3m doesn't render images, right?

**Reply by ymgyt** [reply]
Yes, it might be possible to display images if some settings are configured, but by default, images were not displayed.

**Reply by ymgyt** [reply]
In the [v0.3.1 release](https://github.com/ymgyt/syndicationd/releases/tag/synd-term-v0.3.1), we have added support for invoking a text browser. 
By specifying a command with the `--browser` flag or the `SYND_BROWSER` environment variable, and selecting an entry in the Space, the command `$SYND_BROWSER <entry url>` will be executed.

In the case of a configuration file, you can specify it as follows:

```toml
[feed]
browser = { command = "w3m", args = [] }
```

I'm looking forward to trying out various text browsers from now on.
